documentation/latest/dev/guide.md

```diff
@@ -4,171 +4,38 @@ title: Developer guide
 ```
 
-or
+## Details
+_Parodos_ treats integration tests as unit tests and executes them independently using the Maven _Failsafe_ plugin instead of the classic _Surefire_ plugin.
 
-```shell
-machinectl shell -q flotta@.host /usr/bin/podman pod ps
-```
+The Maven _Surefire_ and _Failsafe_ plugins are two widely used plugins for executing tests during a Maven build. These two plugins differ in the build lifecycle phase in which they operate. While the _Surefire_ plugin runs tests during the test phase, the _Failsafe_ plugin executes tests during the integration-test phase.
 
-if you're running it inside the container.
+The _Failsafe_ plugin is designed to execute integration tests that require more elaborate setup, such as tests that rely on a running application server or database connection. Integration tests run in the same JVM instance as the application, and the plugin includes a default naming convention for identifying integration test classes. _Failsafe_ generates reports that provide insight into the overall health of the application, as well as any integration issues that may exist.
 
-**How can I see the logs produced by the flotta user in the device?**
+To facilitate running integration tests during the Maven build process, _Parodos_ defines a Maven profile named `integrationtest`. The command to execute the integration tests is:
 
-When running podman in rootless mode, the logs from the agent and podman are captured for the flotta user running in the device. Run the following command to see the journalctl logs:
-
-```shell
-machinectl shell -q flotta@.host /usr/bin/journalctl --user
+```bash
+mvn integration-test -P integrationtest
 ```
+You can also specify the server IP and server port by setting the environment variables `SERVERIP` and `SERVERPORT`. By default, the values are `localhost` and `8080`, respectively. For instance, you can run the following command to specify custom values for the server IP and port:
 
-# Device Worker
-
-## CI jobs
-- *Go/build*: Builds current code and checks that unittest are working
-- *Go/Lint*: Checks that code is correctly linted
-- *Go/SecurityScanner*: Checks that the current code is statically checked by
-  [gosec](https://github.com/securego/gosec)
-
-# Developer CLI
-
-### Prerequisites
-- Install the flotta operator as mentioned [here](https://project-flotta.io/documentation/latest/dev/installation.html).
-- Clone the flotta-dev-cli repository and build the project:
-
-```shell
-$ git clone https://github.com/project-flotta/flotta-dev-cli.git
-$ cd flotta-dev-cli
-$ make build
+```bash
+SERVERIP=1.2.3.4 SERVERPORT=9999 mvn integration-test -P integrationtest
 ```
 
-### Running the CLI
-- Use the developer CLI to easily create edge devices and deploy predefine workloads on them:
-
-```shell
-$ ./bin/flotta add device -n device1
-device 'device1' was added
-$ ./bin/flotta add workload -d device1
-workload 'nginx1-21-6-rrxkqkiz' was added to device 'device1'
-```
+## CI Jobs
```

Comment:
Pls  add a section for running integration tests against kind (all the steps from change in source-code to running kind, building the images, pushing to kind and deploying the app before running the tests).
Then running the tests and if there is a hint of how to execute a single test - will be useful.
Next, pls add a section on how to debug failures in the integration tests.

Reply:
@masayag addressed. Can you please take a look?